### PR TITLE
Refactoring runners

### PR DESCRIPTION
**Pull request by badouralix**
Main goal of this PR: run python code with `python`, js code with `node` and so on.

Steps to go:
- [x] Move runners to another dir
- [x] Check if `node` exists on host
- [ ] Add a `--force` flag to bypass the previous check when running on Travis
- [ ] Skip js submission if `node` not found
- [x] Spawn a subprocess running `node`
- [x] Define a protocol to send results to the main process
- [ ] Delegate time measurement to the subprocess

Thus, adding support for any other language will be much easier than binding to some python code.

### CODE REVIEW

**Comment by lypnol** [comment]
@badouralix can you make it retroactive for python submissions ?  
For example you can just change the `submission.py` file to be 
```python
from runners.python import Submission as SubmissionPy
class Submission(SubmissionPy): pass
```
That we'll avoid changing 95 submission files,  
You can keep your changes on `create.py` tool so that new submissions use `runners.python` directly.

File: runners/js.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	def language(self):
+		return 'js'
+
+	def run(self, s):
+		try:
+			output_raw = subprocess.check_output(["node", "-e", self.script, s]).decode()
+			output_rows = output_raw.split('\n')[:-1]
+			if len(output_rows) > 1:
+				print('\n'.join(output_rows[:-1]))
+			return output_rows[-1]
```
Comment (badouralix) [inline runners/js.py:23]: About the protocol to transmit data from the subprocess to the main process: assuming there is only one output written on one single line, `Submission.run` returns the very last line in stdout of the subprocess (any other line is printed out as debug printing.)

**Comment by badouralix** [comment]
Ready to merge this branch! Implementing minor features can be delayed

File: runners/js.py (hunk 1 above)
Comment (lypnol) [reply runners/js.py:23]: Nice approach, you can also add a `rstrip()` to eliminate any trailing spaces or line breaks.

Comment (badouralix) [reply runners/js.py:23]: I'm not sure it's our role to deal with trailing characters. The dev should be allowed to output whatever she wants, formatting it how she intends, without having us interfering. We should only print what we are given, the way it is given.

**Pull request merged**